server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java

```diff
@@ -1450,6 +1455,187 @@ public void shouldConvertPluginToPluginIdOnConfigRepoAsPartOfMigration95() throw
         assertThat(migratedContent, containsString("pluginId=\"json.config.plugin\""));
     }
 
+
+    @Test
+    public void shouldNotSupportedUncesseryMaterialFieldsAsPartOfMigration98() throws Exception {
+        String configXml = "<cruise schemaVersion='98'>" +
+                "<config-repos>\n" +
+                "   <config-repo pluginId=\"json.config.plugin\" id=\"config-repo-1\">\n" +
+                "     <git url=\"https://github.com/tomzo/gocd-json-config-example.git\" dest=\"dest\"/>\n" +
+                "   </config-repo>\n" +
+                "</config-repos>" +
+                "</cruise>";
+
+        String message = "Attribute 'dest' is not allowed to appear in element 'git'.";
+
+        try {
+            migrateXmlString(configXml, 98);
+            fail(String.format("Expected a failure. Reason: Cruise config file with version 98 is invalid. Unable to upgrade. Message:%s", message));
+        } catch (InvocationTargetException e) {
+            assertThat(e.getTargetException().getCause().getMessage(), is(message));
+        }
+    }
+
+    @Test
+    public void shouldRemoveUncesseryMaterialFieldsFromGitMaterialAsPartOfMigration98() throws Exception {
+        String configXml = "<cruise schemaVersion='96'>" +
+                "<config-repos>\n" +
+                "   <config-repo pluginId=\"json.config.plugin\" id=\"config-repo-1\">\n" +
+                "      <git url=\"test-repo\" dest='dest' autoUpdate='false' invertFilter='true' materialName=\"foo\">\n" +
+                "        <filter>\n" +
+                "          <ignore pattern=\"asdsd\" />\n" +
+                "        </filter>\n" +
+                "      </git>" +
+                "   </config-repo>\n" +
+                "</config-repos>" +
+                "</cruise>";
+
+        assertThat(configXml, containsString("<filter>"));
+        assertThat(configXml, containsString("dest='dest'"));
+        assertThat(configXml, containsString("autoUpdate='false'"));
+        assertThat(configXml, containsString("invertFilter='true'"));
+
+        String migratedContent = migrateXmlString(configXml, 97);
+
+        assertThat(migratedContent, not(containsString("<filter>")));
```

Comment:
May be you could convert the migratedContext to Config object using `GoConfigHolder holder = loader.loadConfigHolder(migratedContent);` and then assert that the config-repo object has the right attributes set, say autoUpdate etc.

Reply:
@jyotisingh -- we also need to check whether the attributes are removed from the config.
Will keep the existing assertions and will add new assertions for `null` values for the `attributes` which are not supported on `config-repo` material